### PR TITLE
test(as): add comprehensive tests for type conversion utilities

### PR DESCRIPTION
**Pull request by amery**
## Summary

This PR adds comprehensive tests for type conversion utilities in as.go:

- Generic type conversion functions (As, AsFn)
- Slice conversion functions (SliceAs, SliceAsFn)
- Error conversion functions (AsError, AsErrors)

## Test Coverage

Adds 556 lines of comprehensive test code in `as_test.go` covering:

- `As[T]` and `AsFn[T]` generic type conversions
- `SliceAs[T,S]` and `SliceAsFn[T,S]` slice conversions
- `AsError` and `AsErrors` error interface conversions
- Edge cases: nil inputs, invalid conversions, panic scenarios

## Performance

Benchmarks demonstrate excellent performance with zero allocations for most operations:

```
BenchmarkAs-8         1000000000    0.5717 ns/op    0 B/op    0 allocs/op
BenchmarkAsFn-8       1000000000    0.4926 ns/op    0 B/op    0 allocs/op
BenchmarkSliceAs-8    8860906       155.4 ns/op     160 B/op  1 allocs/op
BenchmarkAsError-8    65276212      17.91 ns/op     0 B/op    0 allocs/op
```

## Testing Approach

Table-driven tests with method-based test cases following established patterns.

## Related

Part of comprehensive test coverage improvement initiative.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit tests for type conversion utilities, covering various input scenarios and edge cases.
  * Introduced benchmark tests to measure performance of conversion functions.
  * Included tests for custom error types and generic behaviours.
  * Added edge case tests to ensure correct error and panic handling.

### CODE REVIEW

**Comment by amery** [comment]
rebased

File: as_test.go [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+	for _, tc := range testCases {
+		t.Run(tc.name, tc.test)
+	}
+}
+
+func TestSliceAs(t *testing.T) {
+	testCases := []sliceAsTestCase{
+		{
+			name:  "mixed types to string",
+			input: []any{"hello", 42, "world", 3.14, "!"},
```
[the remark this answers is not in the document]
Comment (amery) [reply as_test.go]: it should be sorted now

**Review by karasz — APPROVED** [review]
Looks very good to me.
